git-clone/entrypoint.sh

```diff
@@ -32,58 +32,63 @@ git config credential.helper "store --file=.git/credentials"
 echo "https://oauth2:${GITLAB_OAUTH_TOKEN}@${GITLAB_HOST}" > .git/credentials
 git remote add origin $REPOSITORY
 git fetch origin
+git checkout ${BRANCH} || git checkout -b ${BRANCH}
+git submodule init && git submodule update
 
-git config http.proxy http://localhost:8080
-# Note: The proxy will still verify the certificates of
-#       the connection to the git server.
-git config http.sslVerify false
-git config --unset credential.helper
-rm .git/credentials
-
+# If option is set, try to recover from a relevant autosave branch.
 if [ "${GITLAB_AUTOSAVE}" == "1" ] ; then
 
-  # Trying to recover from a relevant autosave branch
+  # Go through available branches and find the appropriate autosave branch
   REMOTES_ORIGIN="remotes/origin/"
   AUTOSAVE_BRANCH_PREFIX="renku/autosave/$JUPYTERHUB_USER"
 
-  GIT_FETCH_OUT=`git fetch && git branch -a`
-  IFS=$'\n' ALL_BRANCHES=($GIT_FETCH_OUT)
+  # Note that the () turn the output into an array
+  ALL_BRANCHES=(`git branch -a `)
   for branch in "${ALL_BRANCHES[@]}"
   do
-      if [[ $branch == *"${REMOTES_ORIGIN}${AUTOSAVE_BRANCH_PREFIX}/${BRANCH}/${COMMIT_SHA:0:7}"* ]] ; then
-          AUTOSAVE_REMOTE_BRANCH=${branch// /}
-      fi
+    # TODO: What if there were are multiple matches here?
+    if [[ $branch == *"${REMOTES_ORIGIN}${AUTOSAVE_BRANCH_PREFIX}/${BRANCH}/${COMMIT_SHA:0:7}"* ]] ; then
+        AUTOSAVE_REMOTE_BRANCH=${branch// /}
+        break
+    fi
   done
 
-fi
+  # If no autosave branch was found, simply reset to the selected commit
+  if [ -z "$AUTOSAVE_REMOTE_BRANCH" ] ; then
+    git reset --hard $COMMIT_SHA
+  else
 
-if [ -z "$AUTOSAVE_REMOTE_BRANCH" ] ; then
-  (git checkout ${BRANCH} || git checkout -b ${BRANCH})
-  git submodule init && git submodule update
-  git reset --hard $COMMIT_SHA
+    IFS='/' read -r -a AUTOSAVE_REMOTE_BRANCH_ITEMS <<< "$AUTOSAVE_REMOTE_BRANCH"
 
-  chown ${USER_ID}:${GROUP_ID} -Rc ${MOUNT_PATH}
-  exit 0
-fi
+    # Check if found autosave branch has valid format, fail otherwise
+    if [ "${#AUTOSAVE_REMOTE_BRANCH_ITEMS[@]}" -lt 7 ] ; then
```

Comment:
@ableuler does this mean that if the `AUTOSAVE_REMOTE_BRANCH_ITEMS` array has less than 7 items then the branch is invalid and the autosave will not be recovered? Why the 7 threshold?

Reply:
The full branch name is split at every occurrence of a `/`, and the resulting array must have 8 or more elements. Consider for example `remotes/origin/renku/autosave/andi/master/f329fe7/f329fe7`